### PR TITLE
Clean up travis.yml and introduce tests for oldest version of PLY supported.

### PR DESCRIPTION
**Pull request by sydneyli**
Cleanup of travis config: 
 * Use env vars instead of Python version
 * Use travis defaults when possible instead of specifying os/distro
 * Add run to test apacheconfig with oldest version of Ply

Bumps up version of Ply to 3.4 since that seems to be the earliest version
available on most distros: https://repology.org/project/python:ply/versions

### CODE REVIEW

**Comment by sydneyli** [comment]
Ah! Discovered and fixed another bug with the older code with this test cleanup (yay for testing!). Failed test here: https://travis-ci.org/etingof/apacheconfig/jobs/633794432

```
  File "/home/travis/virtualenv/python3.8.0/lib/python3.8/site-packages/ply/yacc.py", line 198, in __getitem__
    if n >= 0: return self.slice[n].value
TypeError: '>=' not supported between instances of 'slice' and 'int'
```
In Ply 3.4, the Yacc object don't handle slices correctly for Python 3. This was fixed in [this commit](https://github.com/dabeaz/ply/commit/ee3bb4177786ea343c57208528cc4b5d0cf46918) for the Ply 3.6 release.

This might warrant another point release with the bug fixed!

File: apacheconfig/parser.py [hunk 1]
```diff
@@ -167,7 +167,7 @@ def p_miditem(self, p):
             else:
                 p[0] = p[2]
         else:
-            p[0] = p[1:][0]
+            p[0] = p[1]
```
Comment (etingof) [inline apacheconfig/parser.py]: nit: may be the fix is worth noting in CHANGES?

**Comment by etingof** [comment]
Thank you! o/

**Comment by etingof** [comment]
Is it a good time for another release?

**Comment by m0namon** [comment]
I think so! I'm in the middle of fixing #91 as well, but since that's a cleanup thing I don't think it should block a release.

**Pull request merged**